ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -710,7 +715,14 @@ private boolean verifyJoin(
             ));
           }
 
-          return true;
+          if (!(leftExpression instanceof ColumnReferenceExp)) {
+            throw new KsqlException(String.format(
+                "Invalid join condition: foreign-key table-table joins with expressions "
```

Comment:
We should change the error message: `Invalid join condition: foreign-key table-table joins don't support expressions on the join condition yet.` (conceptually, we can support expression for the left hand side).

Only `rightExpression` _must_ refer to the PK of the right input table and we have already a check for this in place.